editor/src/java/com/defold/control/DefoldStringConverter.java

```diff
@@ -0,0 +1,28 @@
+package com.defold.control;
+
+import clojure.lang.IFn;
+import javafx.util.StringConverter;
+
+public class DefoldStringConverter extends StringConverter<Object> {
+    private final IFn toString;
+    private final IFn fromString;
+
+    public DefoldStringConverter(IFn toString) {
+        this(toString, null);
+    }
+
+    public DefoldStringConverter(IFn toString, IFn fromString) {
+        this.toString = toString;
+        this.fromString = fromString;
+    }
+
+    @Override
+    public String toString(Object o) {
+        return String.valueOf(toString.invoke(o));
```

Comment:
Shouldn't the `toString` `IFn` already return a string? Seems like we're calling `.toString` twice on the object, and maybe a cast would be sufficient?

Reply:
Yes!